x-pack/plugins/security_solution/common/detection_engine/schemas/response/rules_schema.mocks.ts

```diff
@@ -6,13 +6,18 @@
  */
 
 import { DEFAULT_INDICATOR_SOURCE_PATH } from '../../../constants';
+import type {
```

Comment:
> What's the difference between those tests and mocks that are still there?

The tests and mocks in each folder still correspond to the `request` and `response` schemas respectively. It definitely deserves to be reorganized, but I tried to avoid doing reorganization at the same time as the more technical changes. `request/rule_schemas.test.ts` is focused on testing the `createRulesSchema` - we don't have independent tests for `updateRulesSchema` at this point, but it's almost identical to `createRulesSchema`. Similarly the mocks in `request/rule_schemas.mock.ts` are all for the `create` schemas.

Since `request/rule_schemas.ts` is where all of the core schema logic lives, it ends up being the source for the new `response` schemas and replaces `response/rules_schema.ts` - which is simply deleted. However, I kept `response/rules_schema.test.ts` and `response/rules_schema.mock.ts` and updated them to use the new schemas for better assurance that the switch over to the new schema isn't breaking things.

> Can we consolidate all of them under the request folder?

I think we can consolidate by moving most of `request/rule_schemas.ts` to `common/rule_schemas.ts` and better separating the "base" and "type specific" schema logic from create, update, response, etc. This way we can keep request and response schemas organized into their own sections, and also remove the dependency of `response` on `request/rule_schemas.ts`.

Reply:
Great, thank you for clarifying this! Could you please add a few comments to those files to explain this:

> The tests and mocks in each folder still correspond to the request and response schemas respectively. It definitely deserves to be reorganized, but I tried to avoid doing reorganization at the same time as the more technical changes. request/rule_schemas.test.ts is focused on testing the createRulesSchema - we don't have independent tests for updateRulesSchema at this point, but it's almost identical to createRulesSchema. Similarly the mocks in request/rule_schemas.mock.ts are all for the create schemas.

> Since request/rule_schemas.ts is where all of the core schema logic lives, it ends up being the source for the new response schemas and replaces response/rules_schema.ts - which is simply deleted. However, I kept response/rules_schema.test.ts and response/rules_schema.mock.ts and updated them to use the new schemas for better assurance that the switch over to the new schema isn't breaking things.

Reply:
> I think we can consolidate by moving most of request/rule_schemas.ts to common/rule_schemas.ts and better separating the "base" and "type specific" schema logic from create, update, response, etc. This way we can keep request and response schemas organized into their own sections, and also remove the dependency of response on request/rule_schemas.ts.

++ Makes total sense. I added this to https://github.com/elastic/kibana/issues/138606 as a sub-task.